sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1086 @@
+    int16_t pow;     /**< TX power in dBm */
+} ieee802154_phy_conf_t;
+
+/**
+ * @brief Radio ops struct declaration
+ */
+struct ieee802154_radio_ops {
+    /**
+     * @brief Write a frame into the framebuffer.
+     *
+     * This function shouldn't do any checks, so the frame MUST be valid.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] psdu PSDU frame to be sent
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*write)(ieee802154_dev_t *dev, iolist_t *psdu);
+
+    /**
+     * @brief Request the transmission of a preloaded frame
+     *
+     * @ref ieee802154_radio_ops::confirm_transmit MUST be used to finish the
+     * transmission.
+     *
+     * @pre the PHY state is @ref IEEE802154_TRX_STATE_TX_ON and the frame
+     *      is already in the framebuffer.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] flags flags for the transmission
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*request_transmit)(ieee802154_dev_t *dev, ieee802154_tx_flags_t flags);
+
+    /**
+     * @brief Confirmation function for @ref ieee802154_radio_ops::transmit. 
+     *
+     * This function must be called to finish the transmission procedure and
+     * get the transmission status. This function should be called on @ref
+     * IEEE802154_RADIO_CONFIRM_TX_DONE. If no interrupt is available, this
+     * function can be polled.
+     *
+     * @pre the device is on
+     * @pre @ref call to ieee802154_radio_ops::request_transmit was successful.
+     *
+     * @post @ref ieee802154_radio_ops::request_set_trx_state MUST be called
+             in order to set the new transceiver status.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[out] info the TX information. Pass NULL
+     * if the information is not needed. If the radio supports AutoCCA, the
+     * status should indicate transmission done or channel busy. If the radio
+     * supports frame retransmissions, the status should indicate if medium
+     * was busy, no ACK was received or transmission succeeded.
+     *
+     * @return 0 on success
+     * @return -EAGAIN if the transmission has not finished yet.
+     * @return negative errno on error
+     */
+    int (*confirm_transmit)(ieee802154_dev_t *dev, ieee802154_tx_info_t *info);
+
+    /**
+     * @brief Get the lenght of the received PSDU frame.
+     *
+     * @pre the device is on
+     * @pre the radio already received a frame (e.g
+     *      @ref ieee802154_dev_t::cb with @ref IEEE802154_RADIO_RX_DONE).
+     *
+     * @post the frame buffer is still protected against new frame arrivals.
+     *      
+     * @param[in] dev IEEE802.15.4 device descriptor
+     *
+     * @return length of the frame
+     */
+    int (*len)(ieee802154_dev_t *dev);
+
+    /**
+     * @brief Read a PSDU frame from the internal framebuffer of the radio.
```

Comment:
I still find it a bit weird, there is not comment on indicating anything here, although this describes `indication_rx`

Reply:
do you think is enough to update the documentation? or you think the mechanism itself is not clear?

Reply:
Documentation fix is fine for me

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/f9d170103e65e084d331076c7c6d864e63398dfb